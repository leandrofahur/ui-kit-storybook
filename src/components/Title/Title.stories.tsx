import React from "react";
import { Story, Meta } from "@storybook/react";
import Title, { TitleProps } from "./Title";
import styled from "styled-components";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
    title: "Components/Title",
    component: Title,
    argTypes: {},
} as Meta<TitleProps>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: Story<TitleProps> = (args) => <Title {...args} />;

const Container = styled.div`
    display: flex;
    flex-direction: column;

    & > * {
        padding-top: 10px;
    }
`;

export const All = () => {
    return (
        <Container>
            <h5>Typography</h5>
            <Title as="h6">Title</Title>
            <Title as="h5">Title</Title>
            <Title as="h4">Title</Title>
            <Title as="h3">Title</Title>
            <Title as="h2">Title</Title>
            <Title as="h1">Title</Title>
        </Container>
    );
};
