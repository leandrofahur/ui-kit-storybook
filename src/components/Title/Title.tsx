import React, { ElementType } from "react";
import { Heading } from "./Title.styled";

export interface TitleProps {
    /** Props to specify the element as h1, h2, h3 ... h6 */
    as: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
    /** Provides the inner text for the tag */
    children: string;
}

const Title = ({ as, children }: TitleProps) => {
    return <Heading as={as}>{children}</Heading>;
};

export default Title;
