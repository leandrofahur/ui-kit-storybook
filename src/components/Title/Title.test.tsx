import Title from "./Title";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

test("Renders Title with children", () => {
    const { asFragment } = render(<Title as="h1">Hello World</Title>);
    expect(asFragment()).toMatchSnapshot();
});
