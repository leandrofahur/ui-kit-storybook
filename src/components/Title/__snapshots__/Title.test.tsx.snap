// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Title with children 1`] = `
<DocumentFragment>
  .c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  padding-bottom: 20px;
}

<h1
    class="c0"
  >
    Hello World
  </h1>
</DocumentFragment>
`;
